packages/app/src/vis-packs/core/compound/MappedCompoundMatrixVis.tsx

```diff
@@ -0,0 +1,60 @@
+import { MatrixVis } from '@h5web/lib';
+import type {
+  ArrayShape,
+  Dataset,
+  Primitive,
+  PrintableCompoundType,
+  PrintableType,
+} from '@h5web/shared';
+import { capitalize } from 'lodash';
+import ndarray from 'ndarray';
+import { createPortal } from 'react-dom';
+import shallow from 'zustand/shallow';
+
+import MatrixToolbar from '../matrix/MatrixToolbar';
+import { useMatrixConfig } from '../matrix/config';
+import { getCellWidth } from '../matrix/utils';
+import { compoundFormatter } from './utils';
+
+interface Props {
+  dataset: Dataset<ArrayShape, PrintableCompoundType>;
+  value: Primitive<PrintableType>[][];
+  toolbarContainer: HTMLDivElement | undefined;
+}
+
+function MappedCompoundMatrixVis(props: Props) {
+  const { value, dataset, toolbarContainer } = props;
+  const { fields } = dataset.type;
+
+  const fieldNames = Object.keys(fields).map(capitalize);
```

Comment:
I would not use `capitalize` here, as some fields may intentionally start with a lower-case letter, e.g. if they correspond to variable or dataset names. How about capitalising the fields in the mock data instead?